odl/space/fspace.py

```diff
@@ -379,14 +395,6 @@ def __call__(self, x, out=None, **kwargs):
             if ndim == 1:
                 try:
                     out = self._call(x, **kwargs)
-                    if np.ndim(out) == 0 and not scalar_out:
-                        # Don't accept scalar result. A typical situation where
-                        # this occurs is with comparison operators, e.g.
-                        # "return x > 0" which simply gives 'True' for a
-                        # non-empty tuple (in Python 2). We raise TypeError
-                        # to trigger the call with x[0].
```

Comment:
Then it would be good to have an FAQ entry that describes the issue with the 1D function `lambda x: x > 0` returning all ones in Python 2.